engine/src/main/java/org/terasology/telemetry/metrics/Metric.java

```diff
@@ -51,20 +70,22 @@
      * @return a map with key (field's name) and value (field's value).
      */
     public Map<String, ?> getFieldValueMap() {
+        return AccessController.doPrivileged((PrivilegedAction<Map<String, ?>>) () -> {
```

Comment:
A get method with more code than one return statement has a bad name.
This is more like a `createTelemetryFieldValueMap`.